File: 01a/parsing.py
import pandas as pd
import yaml
import xmltodict
## parsing with CSV
csv = pd.read_csv('01a/script.csv')

print("CSV \n", csv.to_string(index=False))

## parsing with JSON
json = pd.read_json('01a/script.json')

print("JSON \n", json.to_string(index=False))
## parsing with YML/YAML
yml_path = '01a/script.yml'
with open (yml_path, 'r') as file:
    yml_data = yaml.safe_load(file)

yml = pd.json_normalize(yml_data)

print("YML \n", yml.to_string(index=False))
## parsing with XML
xml_path = '01a/script.xml'
with open (xml_path, 'r') as file:
    xml_data = xmltodict.parse(file.read())

xml = pd.json_normalize(xml_data)

print("XML \n", xml.to_string(index=False))

File: 01a/parsing.js
const fs = require('fs');
const csv = require('csv-parser');
const yaml = require('js-yaml');
const fastXmlParser = require('fast-xml-parser');

// Parsing CSV
const csvStream = fs.createReadStream('01a/script.csv').pipe(csv());
csvStream.on('data', (row) => console.log("CSV\n", row));

// Parsing YAML
const yaml_data = yaml.load(fs.readFileSync('01a/script.yml', 'utf8'));
console.log('YAML ', yaml_data);
// Parsing XML
const xml_data = fs.readFileSync('01a/script.xml', 'utf8');
const parser = new fastXmlParser.XMLParser();
const xmlParsed = parser.parse(xml_data, { ignoreAttributes: false });
console.log("XML\n", xmlParsed);


// Parsing JSON
const json_data = JSON.parse(fs.readFileSync('./01a/script.json'));
console.log("JSON\n", json_data);
